app/Components/Profile/StrategyOrder.tsx

```diff
@@ -0,0 +1,230 @@
+import React from "react";
+import { View, TouchableOpacity } from "react-native";
+import { IconButton, Text } from "react-native-paper";
+import { sudokuStrategyArray } from "sudokuru";
+import { formatOneLessonName } from "../../Functions/learnedLessons";
+import { PreferencesContext } from "../../Contexts/PreferencesContext";
+
+/**
+ * This is a component for the user to select what order hints for strategies should be given in
+ */
+const StrategyOrder = () => {
+  const { updateStrategyHintOrder, strategyHintOrderSetting } =
+    React.useContext(PreferencesContext);
+
+  const [selectedElement, setSelectedElement] = React.useState(-1);
+
+  // https://stackoverflow.com/questions/40441877/react-native-bulleted-lists-using-flex-wrap
+  // https://stackoverflow.com/questions/39110460/react-native-unordered-style-list
+  /**
+   * A pressable list component that changes border highlight when selected
+   * @param point The content of the element
+   * @param key A key for the element
+   * @param order Specifies which number element this is
+   * @param selected Determines if this element is selected or not
+   * @param bullet The prefix/bullet to use for the element
+   * @returns A JSX element of a hint strategy
+   */
+  const StrategyHintComponent = (
+    point: string,
+    key: number | string,
+    order: number,
+    selected: boolean,
+    bullet = "•"
+  ) => {
+    let borderColor = "grey";
+    if (selected) {
+      borderColor = "#D9A05B";
+    }
+
+    const updateSelectedElement = (order: number) => {
+      if (selected) {
+        return -1;
+      } else {
+        return order;
+      }
+    };
+
+    return (
+      <TouchableOpacity
+        style={{
+          flexDirection: "row",
+          padding: 5,
+          paddingLeft: 10,
+          marginTop: 5,
+          borderRadius: 10,
+          borderWidth: 3,
+          borderColor: borderColor,
+        }}
+        key={key}
+        onPress={() => setSelectedElement(updateSelectedElement(order))}
+      >
+        <Text style={{ fontSize: 14, color: "#025E73", minWidth: 20 }}>
+          {bullet}
+        </Text>
+        <Text style={{ fontSize: 14, paddingLeft: 5, color: "#025E73" }}>
+          {point}
+        </Text>
+      </TouchableOpacity>
+    );
+  };
```

Comment:
Thank you for the feedback. I prefer inline styles for better maintainability. I am not certain on the impacts of using React.memo so I would prefer to avoid it until we reach a point where performance becomes an issue.